Packages/com.unity.inputsystem/InputSystem/Plugins/Switch/SwitchProControllerHID.cs

```diff
@@ -113,12 +129,352 @@ namespace UnityEngine.InputSystem.Switch
+                state.Set(SwitchProControllerHIDInputState.Button.ZR, (buttons0 & 0x80) != 0);
+                state.Set(SwitchProControllerHIDInputState.Button.Minus, (buttons1 & 0x01) != 0);
+                state.Set(SwitchProControllerHIDInputState.Button.Plus, (buttons1 & 0x02) != 0);
+                state.Set(SwitchProControllerHIDInputState.Button.StickL, (buttons1 & 0x04) != 0);
+                state.Set(SwitchProControllerHIDInputState.Button.StickR, (buttons1 & 0x08) != 0);
+
+                var left = false;
+                var up = false;
+                var right = false;
+                var down = false;
+
+                switch (hat)
+                {
+                    case 0:
+                        up = true;
+                        break;
+                    case 1:
+                        up = true;
+                        right = true;
+                        break;
+                    case 2:
+                        right = true;
+                        break;
+                    case 3:
+                        down = true;
+                        right = true;
+                        break;
+                    case 4:
+                        down = true;
+                        break;
+                    case 5:
+                        down = true;
+                        left = true;
+                        break;
+                    case 6:
+                        left = true;
+                        break;
+                    case 7:
+                        up = true;
+                        left = true;
+                        break;
+                }
+
+                state.Set(SwitchProControllerHIDInputState.Button.Left, left);
+                state.Set(SwitchProControllerHIDInputState.Button.Up, up);
+                state.Set(SwitchProControllerHIDInputState.Button.Right, right);
+                state.Set(SwitchProControllerHIDInputState.Button.Down, down);
+
+                return state;
+            }
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size = kSize)]
+        private struct SwitchFullInputReport
+        {
+            public const int kSize = 25;
+            public const byte ExpectedReportId = 0x30;
+
+            [FieldOffset(0)] public byte reportId;
+            [FieldOffset(3)] public byte buttons0;
+            [FieldOffset(4)] public byte buttons1;
+            [FieldOffset(5)] public byte buttons2;
+            [FieldOffset(6)] public byte left0;
+            [FieldOffset(7)] public byte left1;
+            [FieldOffset(8)] public byte left2;
+            [FieldOffset(9)] public byte right0;
+            [FieldOffset(10)] public byte right1;
+            [FieldOffset(11)] public byte right2;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public SwitchProControllerHIDInputState ToHIDInputReport()
+            {
+                ////TODO: calibration curve
```

Comment:
move to jira? what kind of calibration curve do you want to use here?

Reply:
Added ISX-955